src/docs/markdown/build.md

```diff
@@ -50,24 +72,45 @@ To build with plugins, use `--with`:
 
 As you can see, you can customize the versions of plugins with `@` syntax. Versions can be a tag name, commit SHA, or branch.
 
-Cross-platform compilation with `xcaddy` works the same as with the `go` command (see below).
+Cross-platform compilation with `xcaddy` works the same as with the `go` command. For example, to cross-compile for macOS:
 
+<pre><code class="cmd bash">GOOS=darwin xcaddy build</code></pre>
 
-## Cross-platform
+## Build Custom Docker Image
 
-Go programs are easy to compile for other platforms. Just set the `GOOS`, `GOARCH`, and/or `GOARM` environment variables that are different. ([See the go documentation for details.](https://golang.org/doc/install/source#environment))
+Most users deploying production sites will not want to rely on mounting files into a container, but will instead base their own images on `caddy`:
 
-For example, to compile Caddy for Windows when you're not on Windows:
+```Dockerfile
+# note: never use the :latest tag in a production site
+FROM caddy:<version>
 
-<pre><code class="cmd bash">GOOS=windows go build</code></pre>
+COPY Caddyfile /etc/caddy/Caddyfile
+COPY site /srv
+```
 
-Or similarly for Linux ARMv6 when you're not on Linux or on ARMv6:
+#### Adding custom Caddy modules
 
-<pre><code class="cmd bash">GOOS=linux GOARCH=arm GOARM=6 go build</code></pre>
+Caddy is extendable through the use of "modules". See https://caddyserver.com/docs/extending-caddy for full details. You can find a list of available modules on [the Caddy website's download page](https://caddyserver.com/download).
 
-The same works for `xcaddy`. To cross-compile for macOS:
+You can use the `:builder` image as a short-cut to building a new Caddy binary:
 
-<pre><code class="cmd bash">GOOS=darwin xcaddy build</code></pre>
+```Dockerfile
+FROM caddy:<version>-builder AS builder
+
+RUN xcaddy build \
+    --with github.com/caddyserver/nginx-adapter \
+    --with github.com/hairyhenderson/caddy-teapot-module@v0.0.3-0
+
+FROM caddy:<version>
+
+COPY --from=builder /usr/bin/caddy /usr/bin/caddy
+```
+
+Note the second `FROM` instruction - this produces a much smaller image by simply overlaying the newly-built binary on top of the regular `caddy` image.
+
+The [`xcaddy`](https://caddyserver.com/docs/build#xcaddy) tool is used to [build a new Caddy entrypoint](https://github.com/caddyserver/caddy/blob/4217217badf220d7d2c25f43f955fdc8454f2c64/cmd/caddy/main.go#L15..L25), with the provided modules. You can specify just a module name, or a name with a version (separated by `@`). You can also specify a specific version (can be a version tag or commit hash) of Caddy to build from. Read more about [`xcaddy` usage](https://github.com/caddyserver/xcaddy#command-usage).
```

Comment:
Can simplify links to just anchor since they're on this page. I don't think we need to mention the entrypoint stuff or xcaddy syntax here, it should just link to the above section I think.